src/lib/log/util_bug.c

```diff
@@ -69,23 +69,43 @@ tor_set_failed_assertion_callback(void (*fn)(void))
 
 /** Helper for tor_assert: report the assertion failure. */
 void
+__attribute__((format(printf, 5, 6)))
 tor_assertion_failed_(const char *fname, unsigned int line,
-                      const char *func, const char *expr)
+                      const char *func, const char *expr,
+                      const char *fmt, ...)
 {
   char buf[256];
```

Comment:
Is buf large enough for arbitrary printf strings and arguments?

Reply:
We could use tor_asprintf() instead?

Reply:
Doing so in b53a0bc2d1432acb2c8d028461ca6eae2a9b93bc.